ksqldb-engine/src/main/java/io/confluent/ksql/properties/PropertyOverrider.java

```diff
@@ -60,6 +75,18 @@ private static void throwIfInvalidPropertyValues(
               setProperty.getPropertyName(),
               setProperty.getPropertyValue()
           ));
+      if (statement
+          .getSessionConfig()
+          .getConfig(true)
+          .getBoolean(KsqlConfig.KSQL_SHARED_RUNTIME_ENABLED)
+          && !QUERY_LEVEL_CONFIGS.contains(setProperty.getPropertyName())) {
+        throw new KsqlException(String.format("%s is not a settable property at the query"
+                + " level with %s on. Please use ALTER SYSTEM to set %s for the cluster.",
```

Comment:
`KsqlConfig.KSQL_SHARED_RUNTIME_ENABLED` is technically an internal config that users shouldn't be exposed to or really even aware of -- should rephrase to something more along the lines of 


```suggestion
        When shared runtimes are enabled, the config %s can only be set for the entire cluster and all queries currently running in it, and not configurable for individual queries for example by using `SET`.  Please use...
```

or something